### PR TITLE
Port patches from #8624

### PR DESCRIPTION
**Pull request by Owen1212055**
Supersedes https://github.com/PaperMC/Paper/pull/8624

### CODE REVIEW

File: patches/server/0953-Fix-NPE-on-Allay-stopDancing-while-not-dancing.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+     @Override
+     public void stopDancing() {
+         this.getHandle().forceDancing = false;
+-        this.getHandle().jukeboxPos = null;
+-        this.getHandle().setJukeboxPlaying(null, false);
++        // Paper start - Fixing NPE
++        if (this.getHandle().jukeboxPos != null) {
++            this.getHandle().setJukeboxPlaying(this.getHandle().jukeboxPos, false);
++        }
++        // Paper end
```
Comment (Machine-Maker) [inline patches/server/0953-Fix-NPE-on-Allay-stopDancing-while-not-dancing.patch]: Why can't you just use setDancing(false) like the original fix?

Comment (Owen1212055) [reply patches/server/0953-Fix-NPE-on-Allay-stopDancing-while-not-dancing.patch]: Forgot to add that back in, thanks.

Comment (Owen1212055) [reply patches/server/0953-Fix-NPE-on-Allay-stopDancing-while-not-dancing.patch]: Overall switched to simpler logic

Comment (Machine-Maker) [reply patches/server/0953-Fix-NPE-on-Allay-stopDancing-while-not-dancing.patch]: Now the paper comments are too much :)

Comment (Owen1212055) [reply patches/server/0953-Fix-NPE-on-Allay-stopDancing-while-not-dancing.patch]: There's no such thing as TOO many paper comments! 😄

**Review by Machine-Maker — APPROVED** [review]

**Review by lynxplay — APPROVED** [review]

**Review by MiniDigger — APPROVED** [review]

**Pull request merged**